mmgen/models/gans/base_translation_model.py

```diff
@@ -0,0 +1,200 @@
+from copy import deepcopy
+
+import torch.nn as nn
+
+from ..builder import MODELS, build_module
+from .base_gan import BaseGAN
+
+
+@MODELS.register_module()
+class BaseTranslationModel(BaseGAN):
+    """Basic translation Model.
+
+    Args:
+        generator (dict): Config for the generator.
+        discriminator (dict): Config for the discriminator.
+        gan_loss (dict): Config for the gan loss.
+        default_style (str): Default output style.
+        reachable_styles (list[str], optional): Styles that can be generated by
+            the model. Defaults to [].
+        related_styles (list[str], optional): Styles involved in training and
+            testing. Defaults to [].
+        disc_auxiliary_loss (dict): Config for auxiliary loss to
+            discriminator.
+        gen_auxiliary_loss (dict | None, optional): Config for auxiliary loss
+            to generator. Defaults to None.
+        train_cfg (dict): Config for training. Default: None.
+            You may change the training of gan by setting:
+            `disc_steps`: how many discriminator updates after one generator
+            update.
+            `disc_init_steps`: how many discriminator updates at the start of
+            the training.
+        test_cfg (dict): Config for testing. Default: None.
+    """
+
+    def __init__(self,
+                 generator,
+                 discriminator,
+                 gan_loss,
+                 default_style,
+                 reachable_styles=[],
+                 related_styles=[],
+                 disc_auxiliary_loss=None,
+                 gen_auxiliary_loss=None,
+                 train_cfg=None,
+                 test_cfg=None):
+        super().__init__()
+        self.fp16_enabled = False
+        self._default_style = default_style
+        self._reachable_styles = reachable_styles
+        self._related_styles = related_styles
+        assert self._default_style in self._reachable_styles
+        assert set(self._reachable_styles) <= set(self._related_styles)
+
+        # TODO: Building G and D in following way can be used for
+        # Pix2Pix and CycleGAN
+
+        self._gen_cfg = deepcopy(generator)
+        # build style generators
+        self.generators = nn.ModuleDict()
+        for style in self._reachable_styles:
+            self.generators[style] = build_module(generator)
+
+        self._disc_cfg = deepcopy(discriminator)
+        # build style-aware discriminators
```

Comment:
`style-aware` is not accurate. You may use `category-aware` or other names.

Reply:
Is it fine to use `domain discriminators `?

Reply:
I think it's not necessary.
For a GAN-based translation model, discrimintor and generator are all domain-specificed. Otherwise, those modules  would builded by `build_disc/gen_auxiliary`.